kotlin-redux/src/main/kotlin/redux/KotlinEnhancer.kt

```diff
@@ -0,0 +1,34 @@
+package redux
+
+import kotlinext.js.js
+
+fun <S> kotlinEnhancer(next: StoreCreator<S, WrapperAction>): StoreCreator<S, KotlinAction> {
+    return { reducer, preloadedState ->
+        @Suppress("NAME_SHADOWING") // Needed to allow replaceReducer to mutate reducer
+        var reducer = reducer
+        val wrapperReducer: Reducer<S, WrapperAction> = { state, action -> reducer(state, action.action) }
+        val store = next(wrapperReducer, preloadedState)
+object: Store<S, KotlinAction> {
+            override fun getState() = store.getState()
+            override fun dispatch(action: KotlinAction): KotlinAction {
+                @Suppress("UnsafeCastFromDynamic") // Needed because explicit cast causes illegal cast exception
+                store.dispatch(js {
+                    type = action::class.simpleName
+                    this.action = action
+                })
+                return action
+            }
+            override fun subscribe(listener: () -> Unit) = store.subscribe(listener)
+            override fun replaceReducer(nextReducer: Reducer<S, KotlinAction>) {
+                reducer = nextReducer
```

Comment:
Original `replaceReducer` also dispatches an action: https://github.com/reactjs/redux/blob/master/src/createStore.js#L216

Let's reuse it like this:
```
override fun replaceReducer(nextReducer: Reducer<S, KotlinAction>) {
  store.replaceReducer(wrap(nextReducer))
}
```